# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IDBTransaction.h
# Commit: a114b9b8e6ea
# Full Hash: a114b9b8e6eaa23a445a31efc436f4e367f7874f
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-01-10 21:49:10
# Regressor Bug: 1600906
# File Overlap Count: 1
# Description:
#   Bug 1600906 - Convert IDBCursor and BackgroundCursorChild to templates to increase type safety and reduce state. r=dom-workers-and-storage-reviewers,ytausky
#   
#   This also simplifies delegating calls that are dependent on the cursor type.
#   
#   Also reduce dependency on IDBCursor.h by moving enums and type traits to IDBCursorType.h
# ==============================================================================

diff -r cc3cfee4a014 -r a114b9b8e6ea dom/indexedDB/IDBTransaction.h
--- a/dom/indexedDB/IDBTransaction.h	Fri Jan 10 11:07:59 2020 +0000
+++ b/dom/indexedDB/IDBTransaction.h	Fri Jan 10 14:21:18 2020 +0000
@@ -34,7 +34,7 @@
 class StrongWorkerRef;
 
 namespace indexedDB {
-class BackgroundCursorChild;
+class PBackgroundIDBCursorChild;
 class BackgroundRequestChild;
 class BackgroundTransactionChild;
 class BackgroundVersionChangeTransactionChild;
@@ -48,7 +48,6 @@
     : public DOMEventTargetHelper,
       public nsIRunnable,
       public SupportsCheckedUnsafePtr<CheckIf<DiagnosticAssertEnabled>> {
-  friend class indexedDB::BackgroundCursorChild;
   friend class indexedDB::BackgroundRequestChild;
 
  public:
@@ -163,7 +162,7 @@
   indexedDB::BackgroundRequestChild* StartRequest(
       IDBRequest* aRequest, const indexedDB::RequestParams& aParams);
 
-  void OpenCursor(indexedDB::BackgroundCursorChild* aBackgroundActor,
+  void OpenCursor(indexedDB::PBackgroundIDBCursorChild* aBackgroundActor,
                   const indexedDB::OpenCursorParams& aParams);
 
   void RefreshSpec(bool aMayDelete);
@@ -366,10 +365,13 @@
 
   void MaybeNoteInactiveTransaction();
 
+  // TODO consider making private again, or move to the right place
+ public:
   void OnNewRequest();
 
   void OnRequestFinished(bool aRequestCompletedSuccessfully);
 
+ private:
   template <typename Func>
   auto DoWithTransactionChild(const Func& aFunc) const;
 